### PR TITLE
Adjust template to support ceilometer metrics too

### PR DESCRIPTION
**Pull request by csibbitt**
* Using amqpDataSource to determine when to do collectd-specific things as we agreed

### CODE REVIEW

File: roles/smartgateway/templates/deployment.yaml.j2 [hunk 1]
```diff
@@ -23,7 +23,7 @@ spec:
         smart-gateway: '{{ meta.name }}'
     spec:
       containers:
-{% if service_type == 'metrics' %}
+{% if (service_type == 'metrics') and (amqpDataSource == 'collectd') %}
```
Comment (csibbitt) [inline roles/smartgateway/templates/deployment.yaml.j2:26]: This block is for SG3

File: roles/smartgateway/templates/deployment.yaml.j2 [hunk 2]
```diff
@@ -47,15 +47,16 @@ spec:
         volumeMounts:
         - mountPath: /tmp
           name: socket-dir
-{% endif %}
-{% if service_type == 'events' %}
+{% else %}
```
Comment (csibbitt) [inline roles/smartgateway/templates/deployment.yaml.j2:50]: This block is for legacy SG

File: roles/smartgateway/templates/deployment.yaml.j2 [hunk 3]
```diff
@@ -47,15 +47,16 @@ spec:
...
       - name: smart-gateway
         image: '{{ events_container_image_path }}'
```
Comment (csibbitt) [inline roles/smartgateway/templates/deployment.yaml.j2:52]: The name `events_container_image_path` is misleading now. It comes from here: https://github.com/infrawatch/smart-gateway-operator/blob/sg3-bz1805468/roles/smartgateway/tasks/main.yml#L9

I think we should choose a different name, but for the moment I wanted to change as little as possible to help keep the merge simpler.

**Comment by csibbitt** [comment]
Currently untested; honestly not sure when I can get to it. Hoping this implements what @paramite and I discussed, and that it simplifies the upcoming merge.

**Comment by csibbitt** [comment]
See also https://github.com/infrawatch/service-telemetry-operator/pull/97/commits/0261d2b6c68665c3e8029af32918342fdc5efd64 which basically picks the one line from https://github.com/infrawatch/service-telemetry-operator/pull/93/files#diff-33675527951f20f9727fb4be5c84a746R8 that will make this work (hopefully)

**Comment by leifmadsen** [comment]
WRT testing, I'm expecting everything to just merge down to some staging branch that results in the landing of all the SG3 work + Ceilometer Metrics work, with a single set of functional testing for everything set up together.

I don't think we've talked about where that should land right now. Should we create a new staging branch for where this work and the Ceilometer metrics work will eventually land for pre-master functional testing? You've already started the CSV version changes, so we can use that as the next CSV release that includes both of these features.

Thoughts?

**Comment by csibbitt** [comment]
> WRT testing, I'm expecting everything to just merge down to some staging branch that results in the landing of all the SG3 work + Ceilometer Metrics work, with a single set of functional testing for everything set up together.

My sg3-bz1805468 branch in each repo seems like a good candidate for this.

**Comment by leifmadsen** [comment]
> > WRT testing, I'm expecting everything to just merge down to some staging branch that results in the landing of all the SG3 work + Ceilometer Metrics work, with a single set of functional testing for everything set up together.
> 
> My sg3-bz1805468 branch in each repo seems like a good candidate for this.

Fine with me! Sounds like a winner :) (and I'm pretty sure it's already wired up for image building too).

CC: @paramite I guess once your code is ready, then we'll want to switch the base branch from `master` to `sg3-bz1805468` and then resolve any conflicts. Once everything merges down there we can do a functional test for everything at once to make sure we don't have any regressions, while also doing our documentation updates in infrawatch/documentation repo altogether.

**Pull request merged**